p2p/sync.go

```diff
@@ -82,84 +59,61 @@ func (s *syncService) start(ctx context.Context) {
 		if err := ctx.Err(); err != nil {
 			break
 		}
-
 		s.log.Debugw("Continuous iteration", "i", i)
 
 		iterCtx, cancelIteration := context.WithCancel(ctx)
 
-		var err error
-		randHeight, err = s.randomNodeHeight(iterCtx)
-		if err != nil {
-			cancelIteration()
-			if errors.Is(err, errNoPeers) {
-				s.log.Infow("No peers available", "retrying in", retryDuration.String())
-				s.sleep(retryDuration)
-			} else {
-				s.logError("Failed to get random node height", err)
-			}
-			continue
-		}
-
 		var nextHeight int
-		if curHeight, err := s.blockchain.Height(); err == nil { //nolint:govet
+		if curHeight, err := s.blockchain.Height(); err == nil {
 			nextHeight = int(curHeight) + 1
 		} else if !errors.Is(db.ErrKeyNotFound, err) {
 			s.log.Errorw("Failed to get current height", "err", err)
 		}
 
-		blockBehind := randHeight - (nextHeight - 1)
-		if blockBehind <= 0 {
-			s.log.Infow("Random node height is the same or less as local height", " retrying in", retryDuration.String(),
-				"Random node height", randHeight, "Current height", nextHeight-1)
-			cancelIteration()
-			s.sleep(retryDuration)
-			continue
-		}
-
-		s.log.Infow("Start Pipeline", "Random node height", randHeight, "Current height", nextHeight-1, "Start", nextHeight, "End",
-			nextHeight+min(blockBehind, maxBlocks))
+		s.log.Infow("Start Pipeline", "Random node height", randHeight, "Current height", nextHeight-1, "Start", nextHeight)
```

Comment:
`randHeight` will no longer be used, so it will always be 0. Do we need to remove it?

Would like to ask, if we no longer choose a random peer and sync with it, how does the current implementation choose its peers?

I would also recommend adding some sort of check such that if no peers are available, then don't bother trying to sync at all, perhaps retry every certain interval. This is because I've been getting a lot of these messages while attempting to sync when there are no peers:
```
23:53:22.370 14/06/2024 +08:00	ERROR	p2p/sync.go:79	Failed to get block headers parts{"err": "no peers available"}
github.com/NethermindEth/juno/p2p.(*syncService).start
	/Users/han/Documents/Codes/juno/p2p/sync.go:79
github.com/NethermindEth/juno/p2p.(*Service).Run
	/Users/han/Documents/Codes/juno/p2p/p2p.go:212
github.com/NethermindEth/juno/node.(*Node).Run.func3
	/Users/han/Documents/Codes/juno/node/node.go:353
github.com/sourcegraph/conc/panics.(*Catcher).Try
	/Users/han/go/1.22.4/pkg/mod/github.com/sourcegraph/conc@v0.3.0/panics/panics.go:23
github.com/sourcegraph/conc.(*WaitGroup).Go.func1
	/Users/han/go/1.22.4/pkg/mod/github.com/sourcegraph/conc@v0.3.0/waitgroup.go:32
```